File: 0735-asteroid-collision/0735-asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& asteroids) {
        vector<int > st;
        int n=asteroids.size();

        for(int i=0;i<n;i++){
            int right =asteroids[i];
            bool destroy= false;

            while(!st.empty() && st.back()>0 && right<0){
                if(st.back() < abs(right)){
                    st.pop_back();
                    continue;
                }
                else if(st.back() == abs(right)){
                    st.pop_back();
                }
                destroy=true;
                break;
            }
            if(!destroy){
                st.push_back(right);
            }
        }
        return st;
    }
};

File: 0735-asteroid-collision/0735-asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> st = new Stack<>();
        int n= asteroids.length;
        for(int i=0;i<n;i++){
            
            int right = asteroids[i];
            boolean destroy = false;
            while(!st.isEmpty() && st.peek()>0 && right <0){
                if(st.peek() < Math.abs(right)){
                st.pop();
                continue;
                }
                else if(st.peek() == Math.abs(right)){
                    st.pop();

                }
                destroy = true;
                break;
            }
            if(!destroy){
                st.push(right);
            }
        }
        int [] arr= new int[st.size()];
        for(int i=arr.length-1;i>=0;i--){
            arr[i] =st.pop();
        }
        return arr;
    }
}